src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1445,35 +1445,62 @@ private enum GetBytesMode
                     break;
             }
 
-            int curByte = isBigEndian ? length - 1 : 0;
+            int curByte = isBigEndian ? length : 0;
             int increment = isBigEndian ? -1 : 1;
 
             if (bits != null)
             {
-                for (int i = 0; i < bits.Length - 1; i++)
+                if (BitConverter.IsLittleEndian && sign > 0)
                 {
-                    uint dword = bits[i];
+                    ReadOnlySpan<byte> srcBytes = MemoryMarshal.AsBytes(bits.AsSpan(..^1));
 
-                    if (sign == -1)
+                    if (isBigEndian)
+                    {
+                        curByte = length - srcBytes.Length;
+                        Span<byte> destBytes = destination.Slice(curByte, srcBytes.Length);
+                        srcBytes.CopyTo(destBytes);
+                        destBytes.Reverse();
+                    }
+                    else
+                    {
+                        srcBytes.CopyTo(destination);
+                        curByte = srcBytes.Length;
+                    }
+                }
+                else
+                {
+                    for (int i = 0; i < bits.Length - 1; i++)
                     {
-                        dword = ~dword;
-                        if (i <= nonZeroDwordIndex)
+                        uint dword = bits[i];
+
+                        if (sign == -1)
                         {
-                            dword = unchecked(dword + 1U);
+                            dword = ~dword;
+                            if (i <= nonZeroDwordIndex)
+                            {
+                                dword = unchecked(dword + 1U);
+                            }
                         }
-                    }
 
-                    destination[curByte] = unchecked((byte)dword);
-                    curByte += increment;
-                    destination[curByte] = unchecked((byte)(dword >> 8));
-                    curByte += increment;
-                    destination[curByte] = unchecked((byte)(dword >> 16));
-                    curByte += increment;
-                    destination[curByte] = unchecked((byte)(dword >> 24));
-                    curByte += increment;
+                        if (isBigEndian)
+                        {
+                            curByte -= 4;
+                            BinaryPrimitives.WriteUInt32BigEndian(destination.Slice(curByte), dword);
+                        }
+                        else
+                        {
+                            BinaryPrimitives.WriteUInt32LittleEndian(destination.Slice(curByte), dword);
+                            curByte += 4;
+                        }
```

Comment:
I'm missing something... this previously was always writing little endian... why is it now sometimes writing big endian?

Reply:
Oh, I see, increment is 1/-1 based on isBigEndian.